### DISCUSSION TITLE
docker image vulnerability report.

### DISCUSSION THREAD

**Discussion by arjunkrishna** [open-webui/open-webui#1648] [General]
I just came across this in docker about the image vulnerabilities.  one of them is marked critical.

![image](https://github.com/open-webui/open-webui/assets/5271912/0827b98a-4a57-4d8e-ab21-034989eb2a1a)

**Comment by justinh-rahb** [comment]
It's being worked on. So far, updating to non-vulnerable version breaks LiteLLM support in our project entirely however.

**Reply by arjunkrishna** [reply]
do you all do this in your freetime? this is just amazing amount of effort to put in. wow! impressive.

**Reply by justinh-rahb** [reply]
Yes this project is entirely volunteer-driven right now.

**Reply by arjunkrishna** [reply]
maybe, once you add sso, chatgroup/document collection level security then that could be part of a premium paid feature. separation of db for user management/ documents management/ ui settings etc. best of luck.

**Comment by justinh-rahb** [comment]
**Update on LiteLLM Security: Fix Deployed, Testing Underway**

We've made significant progress in addressing the Server-Side Template Injection (SSTI) vulnerability in LiteLLM. The team has developed two potential methods to update LiteLLM to a non-vulnerable version without breaking our functionality, and we've successfully implemented one of them.

A fix is now deployed to the `dev` branch in PR #1653, but we need your help to thoroughly test it before releasing it to production. In the meantime, if you're not using a Hugging Face inference API, your WebUI is safe from exploitation to the best of our current knowledge.

It's essential to note that this vulnerability can only be exploited in specific scenarios:

* An administrator with access to modify the LiteLLM `config.yaml` file adds a vulnerable Hugging Face (HF) model.
* A supply chain attack compromises an existing HF model, which is then used in LiteLLM.

We're committed to ensuring the security and reliability of LiteLLM or at least ensuring continuity of the features from it that we rely upon. We'll continue to monitor the situation and provide updates as we progress.
